src/main/scala/firrtl/options/ExecutionOptionsManager.scala

```diff
@@ -33,12 +34,53 @@ class ExecutionOptionsManager(
   /** Show usage and exit */
   def showUsageAsError(): Unit = parser.showUsageAsError()
 
+  /* Mechanism to collect the options of  */
+  private var registeredLibraries = scala.collection.mutable.ArrayBuffer[RegisteredLibrary]()
+  private def addRegisteredOptions(): Unit = {
+    val iter = ServiceLoader.load(classOf[RegisteredLibrary]).iterator()
+    while (iter.hasNext) {
+      val lib = iter.next()
+      registeredLibraries += lib
+      parser.note(lib.name)
+      lib.addOptions(parser)
+    }
+  }
+
+  /* Mechanism to collect the options of individual transforms */
+  private var registeredTransforms = scala.collection.mutable.ArrayBuffer[RegisteredTransform]()
+  private def addTransformOptions(): Unit = {
+    val iter = ServiceLoader.load(classOf[RegisteredTransform]).iterator()
+    parser.note("FIRRTL Transform Options")
+    while (iter.hasNext) {
+      val tx = iter.next()
+      registeredTransforms += tx
+      tx.addOptions(parser)
+    }
+  }
+
   /** The [[AnnotationSeq]] generated from command line arguments
     *
     * This requires lazy evaluation as subclasses will mixin new command
     * line options via methods of [[ExecutionOptionsManager.parser]]
     */
-  lazy implicit final val options: AnnotationSeq = parser
-    .parse(args, annotations)
-    .getOrElse(throw new FIRRTLException("Failed to parse command line options"))
+  lazy implicit final val options: AnnotationSeq = {
+    addTransformOptions
+    addRegisteredOptions
+    parser
+      .parse(args, annotations)
+      .getOrElse(throw new FIRRTLException("Failed to parse command line options"))
+  }
+
+  /* A hidden option that exposes what libraries and transforms are
+   * registered. This is purely for debugging. */
+  parser.opt[Unit]("show-registrations")
+    .hidden()
+    .action{ (_, c) =>
+      println(s"""|Registered Libraries:
```

Comment:
println or should this be logger.info? logger.warn?

Reply:
I think this is safe to leave in as it is. I could wrap this in a logger, but this is really information that I want to get to the command line for debugging before a logger may exist and I want this to happen even if a logger doesn't exist or I have the incorrect log level. 

I.e., I think I can wrap the scopt `parse` call with a logger, but I'd want to pass in an explicit log level which would seem to violate the motivation for having a logger at all.